ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/chain/DefaultMutableBlockchain.java

```diff
@@ -355,7 +363,17 @@ private BlockAddedEvent handleChainReorg(
         });
     clearIndexedTransactionsForBlock(updater, removedTransactions);
 
-    // Update tracked forks
+    if (!isFastSync) {
+      updateForksForReorg(updater, newChainHead, oldChainHead);
```

Comment:
I _think_ we're tracking the fork heads so that we can go back and clean them up at some point.  In which case, it seems useful to keep that set accurate.

Reply:
Yeah I was initially aiming to limit the number of headers we'd encounter without world states but #824 handles that case so probably is better to track forks made during fast sync too.